tensorflow/python/kernel_tests/cholesky_op_test.py

```diff
@@ -87,6 +88,26 @@ def testEmpty(self):
     self._verifyCholesky(np.empty([0, 2, 2]))
     self._verifyCholesky(np.empty([2, 0, 0]))
 
+class CholeskyGradTest(tf.test.TestCase):
+
+  def testCholeskyGrad(self):
+    np.random.seed(0)
+    backprop_block_size = 32    
+    shapes = ( (elem, 2*elem) for elem in [1,2,10,backprop_block_size+1,2*backprop_block_size+1] )
+    dtypes = (tf.float32, tf.float64)
+    with self.test_session(use_gpu=False):
+      for shape in shapes:             
+        for dtype in dtypes:
+          x = tf.constant( np.random.rand(shape[0]) )
+          R = tf.constant( np.random.randn(shape[0],shape[1])  )
+          e = tf.matmul( tf.diag( x ), R )
+          K = tf.matmul( e, tf.transpose(e) ) / shape[0] #This gives a positive definite matrix.
```

Comment:
OK, generally we want individual tests to run in 30-45 seconds without optimization. You might want to break your test into multiple ones instead of looping over the combination of all parameters within a single test.